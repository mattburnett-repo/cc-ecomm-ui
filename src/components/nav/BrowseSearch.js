export default function BrowseSearch () {
    return (
        <div role="presentation" aria-label="browse-search">
            <div role="presentation" aria-label="product-categories">select product categories</div>
            <input aria-label="search-terms" placeholder="Enter search terms"></input>
            <button aria-label="search-button">Search</button>
        </div>
    )           
}