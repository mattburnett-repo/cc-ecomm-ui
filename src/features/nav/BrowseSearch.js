import { useSelector } from "react-redux";

import CheckLoginStatus from "../../utils/CheckLoginStatus";
import BrowseSearchDisplay from '../../components/nav/BrowseSearchDisplay';

import { selectProductCategories } from "../productCategory/productCategorySlice";
// eslint-disable-next-line 
import { getProducts, getProductsByCategory } from "../products/productsSlice";

export default function BrowseSearch () {
    CheckLoginStatus()

    // TODO: need to figure out how to get dispatch() to work from here
    function handleChange(e){
        let categoryId = e.target.value

        // eslint-disable-next-line
        if(categoryId == 0) {
            alert('browseSearch select all products. \nneed to figure out how to get dispatch() to work from here')
        } else {
            alert('browseSearch select products for category: ' + categoryId + '. \nneed to figure out how to get dispatch() to work from here')
        }
    }

    function handleSearch(e){
        e.preventDefault()
        // let searchTerms = e.target.value
        alert('handleSearch for search term/s ' + e.target.searchTerms.value + '. \nneed to figure out how to get dispatch() to work from here')
    }
        
    function handlers() {
        return {
            handleChange: handleChange,
            handleSearch: handleSearch,
        }
    }

    const productCategories = useSelector(selectProductCategories)

    return <BrowseSearchDisplay  data={ productCategories } handlers={handlers()}/>
}