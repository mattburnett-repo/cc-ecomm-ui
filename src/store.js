import { configureStore } from '@reduxjs/toolkit';

import authReducer from './features/auth/authSlice';
import productCategoryReducer from './features/productCategory/productCategorySlice'
import productsListingReducer from './features/products/productsSlice'
import ordersReducer from './features/orders/ordersSlice'
import cartsReducer from './features/carts/cartsSlice'

function rootReducer() {
    return {
        reducer: {
            auth: authReducer,
            productCategories: productCategoryReducer,
            orders: ordersReducer,
            carts: cartsReducer,
            products: productsListingReducer,
        }    
    }
}

const store = configureStore(rootReducer());

export default store;