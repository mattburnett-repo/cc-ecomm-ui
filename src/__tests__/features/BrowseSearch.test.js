import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';

import { Provider } from 'react-redux';
import store from '../../store'

import BrowseSearch from '../../features/nav/BrowseSearch';

describe('<BrowseSearch /> feature tests', () => {
    beforeEach(() => render(<Provider store={store}><BrowseSearch /></Provider>));

    it('displays product categories', () => {
        const theVal = screen.getByRole('presentation', {name: /product-categories/i});
        expect(theVal.length).toBe(4);
    })
    it('selects a product category', () => {
        fireEvent.change(screen.getByRole('presentation', {name: /product-categories/i}), { target: { value: 2 } })
        let options = screen.getByRole('presentation', {name: /product-categories/i});
        expect(options[0].selected).toBeFalsy();
        expect(options[1].selected).toBeFalsy();
        expect(options[2].selected).toBeFalsy();
        expect(options[3].selected).toBeTruthy();
    })

    it('accepts search text', () => {
        const theElement = screen.getByRole('textbox', {name: /search-terms/i});
        userEvent.type(theElement, 'here is some search text');
        
        expect(theElement).toHaveValue('here is some search text');
    });
    it('clicks the search button', () => {
        const theVal = screen.getByRole('button', {name: /search/i});
        fireEvent.click(theVal);
    });
    it('creates a snapshot', () => {
        const tree = renderer.create(<Provider store={store}><BrowseSearch /></Provider>).toJSON();
        expect(tree).toMatchSnapshot();  
    })
}); // end feature
