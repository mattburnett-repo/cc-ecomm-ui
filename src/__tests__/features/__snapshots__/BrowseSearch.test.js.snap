// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BrowseSearch /> feature tests creates a snapshot 1`] = `
<div
  aria-label="browse-search"
  role="presentation"
>
  <div
    aria-label="browse"
    role="presentation"
  >
    <label
      htmlFor="product-categories"
    >
      Product Categories: 
    </label>
    <select
      aria-label="product-categories"
      defaultValue="0"
      id="product-categories"
      name="productCategories"
      onChange={[Function]}
      role="presentation"
    >
      <option
        value="0"
      >
        Show all products
      </option>
    </select>
  </div>
  <div
    aria-label="search"
    role="presentation"
  >
    <form
      method="post"
      onSubmit={[Function]}
    >
      <div>
        <label
          htmlFor="searchTerms"
        >
          Search: 
        </label>
        <input
          aria-label="search-terms"
          id="searchTerms"
          placeholder="Enter search terms"
        />
      </div>
      <div>
        <button
          aria-label="search-button"
          type="submit"
        >
          Search
        </button>
      </div>
    </form>
  </div>
</div>
`;
